xbmc/interfaces/legacy/ListItem.cpp

```diff
@@ -479,6 +479,20 @@ namespace XBMCAddon
       }
       else if (strcmpi(type, "music") == 0)
       {
+        std::string type;
+        for (InfoLabelDict::const_iterator it = infoLabels.begin(); it != infoLabels.end(); ++it)
+        {
+          String key = it->first;
+          StringUtils::ToLower(key);
+          const InfoLabelValue& alt = it->second;
+          const String value(alt.which() == first ? alt.former() : emptyString);
+
+          if (key == "mediatype")
+          {
+            if (CMediaTypes::IsValidMediaType(value))
+              type = value;
```

Comment:
You can also already set the mediatype here (setType), that way you can remove it from below.